rskj-core/src/test/java/co/rsk/peg/BridgeSupportSvpTest.java

```diff
@@ -662,11 +638,296 @@ private void assertInputHasExpectedScriptSig(TransactionInput input, Script rede
+                .withBridgeConstants(bridgeMainNetConstants)
+                .withProvider(bridgeStorageProvider)
+                .withActivations(allActivations)
+                .withFederationSupport(federationSupport)
+                .withFeePerKbSupport(feePerKbSupport)
+                .withEventLogger(bridgeEventLogger)
+                .withExecutionBlock(rskExecutionBlock)
+                .build();
+
+            // act
+            for (BtcECKey key : PROPOSED_FEDERATION_SIGNERS_KEYS) {
+                List<byte[]> signerEncodedSignatures = generateSignerEncodedSignatures(key, svpSpendTxSigHashes);
+                bridgeSupport.addSignature(key, signerEncodedSignatures, svpSpendTxCreationHash);
+            }
+
+            // assert
+            for (BtcECKey key : PROPOSED_FEDERATION_SIGNERS_KEYS) {
+                assertFederatorDidNotSignInputs(svpSpendTx.getInputs(), svpSpendTxSigHashes, key);
+            }
+
+            assertAddSignatureWasNotLogged();
+            assertSvpSpendTxWFSWasNotRemoved();
+        }
+
+        private void assertFederatorDidNotSignInputs(List<TransactionInput> inputs, List<Sha256Hash> sigHashes, BtcECKey key) {
+            for (TransactionInput input : inputs) {
+                Sha256Hash sigHash = sigHashes.get(inputs.indexOf(input));
+                assertFalse(BridgeUtils.isInputSignedByThisFederator(key, sigHash, input));
+            }
+        }
+
+        private void assertAddSignatureWasNotLogged() {
+            assertEquals(0, logs.size());
+        }
+
+        private void assertSvpSpendTxWFSWasNotRemoved() {
+            assertTrue(bridgeStorageProvider.getSvpSpendTxWaitingForSignatures().isPresent());
+        }
+
+        @Test
+        void addSignature_forSvpSpendTx_shouldAddProposedFederatorsSignatures() throws Exception {
+            // act
+            for (BtcECKey key : PROPOSED_FEDERATION_SIGNERS_KEYS) {
+                List<byte[]> signerEncodedSignatures = generateSignerEncodedSignatures(key, svpSpendTxSigHashes);
+                bridgeSupport.addSignature(key, signerEncodedSignatures, svpSpendTxCreationHash);
+            }
+
+            // assert
+            for (BtcECKey key : PROPOSED_FEDERATION_SIGNERS_KEYS) {
+                assertFederatorSigning(
+                    svpSpendTxCreationHash.getBytes(),
+                    svpSpendTx.getInputs(),
+                    svpSpendTxSigHashes,
+                    proposedFederation,
+                    key
+                );
+            }
+            assertLogReleaseBtc(svpSpendTxCreationHash, svpSpendTx);
+            assertLogsSize(PROPOSED_FEDERATION_SIGNERS_KEYS.size() + 1); // proposedFedSigners size for addSignature, +1 for release btc
+            assertFalse(bridgeStorageProvider.getSvpSpendTxWaitingForSignatures().isPresent());
+        }
+
+        @Test
+        void addSignature_forNormalPegout_whenSvpIsOngoing_shouldAddJustActiveFederatorsSignaturesToPegout() throws Exception {
```

Comment:
What happens if the signer is an active member of both feds, the active federation and the proposed federation? Will it sign both pegouts in the same rsk tx?

Reply:
wdym with both pegouts? the tx being signed is the one that has the given `rskTxHash` 🤔